doc/manual/generics.txt

```diff
@@ -425,10 +429,25 @@ types, thus allowing you to encode implementation detail types:
                   # and it must be a numeric sequence
 
 As seen in the previous examples, you can refer to generic concepts such as
-Enumerable[T] just by their short name. Much like the regular generic types,
+`Enumerable[T]` just by their short name. Much like the regular generic types,
 the concept will be automatically instantiated with the bind once auto type
 in the place of each missing generic param.
 
+Please note that generic concepts such as `Enumerable[T]` can be matched
+against concrete types such as `string`. Nim doesn't require the concept
+type to have the same number of parameters as the type being matched.
+In order to express such a requirement, you'll need to rely on a type
+mapping operator such a `genericHead` or `stripGenericParams` within the
+concept body:
+
+.. code-block:: nim
+  import future, typetraits
+
+  type
+    Functor[A] = concept f
+      f.value is A
+      map(f, A -> T1) is genericHead(f.type)[T1]
```

Comment:
I'll expand this or come up with a simpler example. `Functor` is a well-known concept from Haskell, and that's why I considered it a clear-enough example.